File: remove_bg.py
from PIL import Image

def remove_white_bg(img_path):
    img = Image.open(img_path)
    img = img.convert("RGBA")
    datas = img.getdata()

    newData = []
    # threshold for white
    for item in datas:
        # Check if the pixel is white or very close to white
        if item[0] > 230 and item[1] > 230 and item[2] > 230:
            newData.append((255, 255, 255, 0)) # transparent
        else:
            newData.append(item)

    img.putdata(newData)
    img.save(img_path, "PNG")
    print("Background removed successfully!")

remove_white_bg("src/assets/logo.png")

File: remove-bg.mjs
import { Jimp } from 'jimp';

async function removeBackground() {
  try {
    const imagePath = './src/assets/logo.png';
    console.log('Reading image:', imagePath);
    const image = await Jimp.read(imagePath);
    
    const width = image.bitmap.width;
    const height = image.bitmap.height;
    
    // We will consider pixels close to white as background
    const threshold = 230; 
    
    image.scan(0, 0, width, height, function(x, y, idx) {
      const r = this.bitmap.data[idx + 0];
      const g = this.bitmap.data[idx + 1];
      const b = this.bitmap.data[idx + 2];
      
      // If the pixel is mostly white, we make it transparent
      if (r > threshold && g > threshold && b > threshold) {
        // Make it completely transparent
        this.bitmap.data[idx + 3] = 0;
      }
    });
    
    await image.writeAsync(imagePath);
    console.log('Background removed successfully!');
  } catch (err) {
    console.error('Error removing background:', err);
  }
}

removeBackground();
